src/pal/src/misc/sysinfo.cpp

```diff
@@ -103,11 +103,19 @@ PAL_GetLogicalCpuCountFromOS()
     int nrcpus = 0;
 
 #if HAVE_SYSCONF
-    nrcpus = sysconf(_SC_NPROCESSORS_ONLN);
+#if defined(_ARM_) || defined(_ARM64_)
```

Comment:
A nit - it seems that `#define` a symbol to _SC_NPROCESSORS_CONF or _SC_NPROCESSORS_ONLN based on the architecture and then having the rest of the code common would be better from the maintenance point of view and make the code shorter.